File: gluecodium/src/test/resources/smoke/properties/output/android/com/example/smoke/CachedProperties.java
/*

 *
 */

package com.example.smoke;

import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.List;

public final class CachedProperties extends NativeBase {

    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected CachedProperties(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }

    private static native void disposeNativeHandle(long nativeHandle);


    private List<String> cache_getCachedProperty;
    private boolean is_cached_getCachedProperty = false;


    @NonNull
    public List<String> getCachedProperty() {
        if (!is_cached_getCachedProperty) {
            cache_getCachedProperty = getCachedProperty_private();
            is_cached_getCachedProperty = true;
        }
        return cache_getCachedProperty;
    }

    private native List<String> getCachedProperty_private();


    private List<String> cache_getInternalCachedProperty;
    private boolean is_cached_getInternalCachedProperty = false;


    @NonNull
    List<String> getInternalCachedProperty() {
        if (!is_cached_getInternalCachedProperty) {
            cache_getInternalCachedProperty = getInternalCachedProperty_private();
            is_cached_getInternalCachedProperty = true;
        }
        return cache_getInternalCachedProperty;
    }

    private native List<String> getInternalCachedProperty_private();


    private static byte[] cache_getStaticCachedProperty;
    private static boolean is_cached_getStaticCachedProperty = false;


    @NonNull
    public static byte[] getStaticCachedProperty() {
        if (!is_cached_getStaticCachedProperty) {
            cache_getStaticCachedProperty = getStaticCachedProperty_private();
            is_cached_getStaticCachedProperty = true;
        }
        return cache_getStaticCachedProperty;
    }

    private static native byte[] getStaticCachedProperty_private();


    private static byte[] cache_getInternalStaticCachedProperty;
    private static boolean is_cached_getInternalStaticCachedProperty = false;


    @NonNull
    static byte[] getInternalStaticCachedProperty() {
        if (!is_cached_getInternalStaticCachedProperty) {
            cache_getInternalStaticCachedProperty = getInternalStaticCachedProperty_private();
            is_cached_getInternalStaticCachedProperty = true;
        }
        return cache_getInternalStaticCachedProperty;
    }

    private static native byte[] getInternalStaticCachedProperty_private();




}

File: gluecodium/src/test/resources/smoke/properties/output/android-kotlin/com/example/smoke/CachedProperties.kt
/*

 *
 */

@file:JvmName("CachedPropertiesExtensions")


package com.example.smoke

import com.example.NativeBase

class CachedProperties : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    val cachedProperty: List<String>
        get() {
            if (!is_cached_cachedProperty) {
                cache_cachedProperty = getCachedProperty_private()
                is_cached_cachedProperty = true
            }

            return cache_cachedProperty!!
        }

    private var is_cached_cachedProperty = false
    private var cache_cachedProperty: List<String>? = null
    external private fun getCachedProperty_private() : List<String>


    internal val internalCachedProperty: List<String>
        @JvmSynthetic get() {
            if (!is_cached_internalCachedProperty) {
                cache_internalCachedProperty = getInternalCachedProperty_private()
                is_cached_internalCachedProperty = true
            }

            return cache_internalCachedProperty!!
        }

    private var is_cached_internalCachedProperty = false
    private var cache_internalCachedProperty: List<String>? = null
    external private fun getInternalCachedProperty_private() : List<String>





    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
        @JvmStatic val staticCachedProperty: ByteArray
            get() {
                if (!is_cached_staticCachedProperty) {
                    cache_staticCachedProperty = getStaticCachedProperty_private()
                    is_cached_staticCachedProperty = true
                }

                return cache_staticCachedProperty!!
            }

        @JvmStatic private var is_cached_staticCachedProperty = false
        @JvmStatic private var cache_staticCachedProperty: ByteArray? = null
        @JvmStatic external private fun getStaticCachedProperty_private() : ByteArray


        @JvmStatic internal val internalStaticCachedProperty: ByteArray
            @JvmSynthetic get() {
                if (!is_cached_internalStaticCachedProperty) {
                    cache_internalStaticCachedProperty = getInternalStaticCachedProperty_private()
                    is_cached_internalStaticCachedProperty = true
                }

                return cache_internalStaticCachedProperty!!
            }

        @JvmStatic private var is_cached_internalStaticCachedProperty = false
        @JvmStatic private var cache_internalStaticCachedProperty: ByteArray? = null
        @JvmStatic external private fun getInternalStaticCachedProperty_private() : ByteArray


    }
}


@JvmSynthetic
fun CachedProperties.Companion.getStaticCachedProperty(): ByteArray = CachedProperties.staticCachedProperty

@JvmSynthetic
internal fun CachedProperties.Companion.getInternalStaticCachedProperty(): ByteArray = CachedProperties.internalStaticCachedProperty

